### PR TITLE
Qt: Adds Governance tab

### PR DESCRIPTION
**Pull request by 5tefan**
Towards #4337, implementing DMT-like Governance Functionality.

### CODE REVIEW

**Comment by 5tefan** [comment]
Views of the new functionality:

Screen Shot 2021-08-20 at 01 07 19

Screen Shot 2021-08-20 at 01 07 00

Questions:
- What columns to display? Probably include vote counts/ratio. Is showing url necessary with right click to open url exists?
- Looking a bit simper than Masternode list, not subscribing to updates from core: just periodically refresh, or include a refresh button?
- Show proposals only?

**Comment by UdjinM6** [comment]
Nice! 👍 

> Questions:
> 
> * What columns to display? Probably include vote counts/ratio. Is showing url necessary with right click to open url exists?

Would be nice to have "Voting status" implemented in a similar way it's done in DMT imo. Also, cycles (e.g. current/total) and total amount might be useful in this list. URL, yes/no/abstain counts etc. can be shown in a separate panel like it's done in DMT (i.e. you select a proposal and this populates "Details" panel below).

> * Looking a bit simper than Masternode list, not subscribing to updates from core: just periodically refresh, or include a refresh button?

The list is small here (10s comparing to 1000s in mn list), periodic updates shouldn't be ok I think (assuming they won't cause issues with "Details" panel I mentioned above).

> * Show proposals only?

Yeah, I think it would be better to start with proposals first. We can get back to triggers later when we polish proposals list(/vote?) functionality.

File: src/qt/governancelist.cpp [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_ACTIVE, columnActiveWidth);
+
+    ui->tableWidgetGovernances->setContextMenuPolicy(Qt::CustomContextMenu);
+    ui->tableWidgetGovernances->sortItems(COLUMN_CREATION, Qt::DescendingOrder);
+
+
+    ui->filterLineEdit->setPlaceholderText(tr("Filter by Name"));
+
+    timer = new QTimer(this);
+    connect(timer, SIGNAL(timeout()), this, SLOT(updateGovernanceListScheduled()));
```
Comment (UdjinM6) [inline src/qt/governancelist.cpp:76]: This should make linter happy again
```suggestion
    connect(timer, &QTimer::timeout, this, &GovernanceList::updateGovernanceListScheduled);
```

File: src/qt/optionsdialog.cpp [hunk 2]
```diff
@@ -238,7 +238,7 @@ void OptionsDialog::setModel(OptionsModel *_model)
     connect(ui->threadsScriptVerif, static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged), this, &OptionsDialog::showRestartWarning);
     /* Wallet */
     connect(ui->showMasternodesTab, &QCheckBox::clicked, this, &OptionsDialog::showRestartWarning);
-    connect(ui->spendZeroConfChange, &QCheckBox::clicked, this, &OptionsDialog::showRestartWarning);
+    connect(ui->showGovernanceTab, &QCheckBox::clicked, this, &OptionsDialog::showRestartWarning);
```
Comment (UdjinM6) [inline src/qt/optionsdialog.cpp:241]: 👀 👍

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Comment by 5tefan** [comment]
Formatting addressed. Thanks, that `clang-format-diff.py` script is super convenient!

**Review by PastaPastaPasta — APPROVED** [review]
ACK for squash merge

**Comment by PastaPastaPasta** [comment]
Things for future changes: 
Add: Yes count, No count, Abstain Count, Absolute Yes coin, url
Remove: Hash
Replace Start, End dates with computed "Payments Remaining"

We should consider adding a pop-up warning before opening the proposal URL

**Pull request merged**